plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProject.java

```diff
@@ -118,17 +125,33 @@ public void execute(IProgressMonitor monitor) throws InvocationTargetException,
       throw new InvocationTargetException(ex);
     }
 
-    addAppEngineFacet(newProject, subMonitor.newChild(4));
+    addAppEngineFacet(newProject, subMonitor.newChild(6));
 
     if (config.getUseMaven()) {
-      enableMavenNature(newProject, subMonitor.newChild(2));
+      enableMavenNature(newProject, subMonitor.newChild(4));
+      BuildPath.addMavenLibraries(newProject, config.getAppEngineLibraries(), subMonitor.newChild(5));
     } else {
       addJunit4ToClasspath(newProject, subMonitor.newChild(2));
-    }
+      IJavaProject javaProject = JavaCore.create(newProject);
+      
+      List<Library> libraries = config.getAppEngineLibraries();
+      // todo duplicates some code in CloudLibrariesPage
+      SortedSet<LibraryFile> masterFiles = new TreeSet<>();
```

Comment:
This code seems like it should be hidden in BuildPath#addNativeLibrary()`.

Reply:
done